app/controllers/declaration/AdditionalProcedureCodesController.scala

```diff
@@ -205,8 +206,8 @@ class AdditionalProcedureCodesController @Inject()(
         val continueToCommodityDetails = navigator.continueTo(mode, routes.CommodityDetailsController.displayPage(_, itemId))
 
         val model = maybeModel.fold(request.cacheModel)(identity)
-        if (SupervisingCustomsOfficeHelper.isConditionForAllProcedureCodesNotVerified(model)) Future.successful(continueToCommodityDetails)
-        else SupervisingCustomsOfficeHelper.resetCache(this, model).map(_ => continueToCommodityDetails)
+        if (isConditionForAllProcedureCodesVerified(model)) resetCache(this, model).map(_ => continueToCommodityDetails)
```

Comment:
Nit: just for clarity, now you've removed the class prefix of `resetCache` perhaps its name could be more specific, like `clearSupervisingCustomsOfficeInCache`? (`resetCache` sounds like a global reset of the entire cache) :slightly_smiling_face:

Reply:
:+1: Indeed, I was also thinking to change it.